File: BookManagementSystem/PhysicalBook.java
package BookManagementSystem;

// The PhysicalBook class must inherit the Book class and have the following additional attributes:
    // weight: a double representing the weight of the physical book.
    // shipping: a double representing the shipping price of the physical book.


public class PhysicalBook extends Book{
    private double weight;
    private double shipping;
    private int quantity;

    // constructor
    public PhysicalBook(String title, String author, String isbn, double price, double weight, int quantity, double shipping){
        super(title, author, isbn, price);
        this.weight = weight;
        this.shipping = shipping;
        this.quantity = quantity;
    }

    // method to calculate the total price
    @Override
    public double calculatePriceTotal() {
        return price + shipping;
    }

    // getters and setters

    // get and set weight
    public double getWeight(){
        return weight;
    }
    public void setWeight(double weight){
        this.weight = weight;
    }

    // get and set shipping
    public double getShipping(){
        return shipping;
    }
    public void setShipping(double shipping){
        this.shipping = shipping;
    }

    // get and set quantity
    public int getQuantity(){
        return quantity;
    }
    public void setQuantity(int quantity){
        this.quantity = quantity;
    }

    // method to calculate the total weight
    public double calculateWeightTotal() {
        return weight;
    }

    // method to calculate the total shipping
    public double calculateShippingTotal() {
        return shipping;
    }
}

File: BMSRust/src/PhysicalBook.rs
mod Book;
use Book::Book;

pub struct EBook{
    book: Book,
    weight: f64,
    shipping: f64
}

pub impl PhysicalBook{
    pub fn new(title: String, author: String, isbn: String, price: f64, weight: f64, shipping: f64) -> PhysicalBook{
        PhysicalBook{
            book: Book{
                title,
                author,
                isbn,
                price,
            },
            weight,
            shipping
        }
    }

    // getters and setters

    pub fn calculate_price_total(&self) -> f64 { // method to calculate the total price of the book
        self.book.price + self.shipping
    }

    // calculate method for weight
    pub fn calculate_weight(&self) -> &f64{
        &self.weight
    }

    // getter for weight
    pub fn get_weight(&self) -> f64 {
        self.weight
    }

    // setter for weight
    pub fn set_weight(&mut self, weight: f64){
        self.weight = weight;
    }

    // getter for shipping
    pub fn get_shipping(&self) -> f64 {
        self.shipping
    }

    // setter for shipping
    pub fn set_shipping(&mut self, shipping: f64){
        self.shipping = shipping;
    }
}